graphing/src/cloudevolution/views.py

```diff
@@ -94,7 +94,54 @@ def expt_name(request, experiment):
 
 	return render(request, "experiment.html", context)
 
+def dilutions(request, experiment):
+	sidebar_links, subdir_log = file_scan('expt')
+	vial_count = range(0, 16)
+	expt_dir, expt_subdir = file_scan(experiment)
+	rootdir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+	evolver_dir = rootdir + '/experiment'
+	pump_cal = evolver_dir + "/%s/pump_cal.txt" % (expt_subdir[0])
+
+	cal = np.genfromtxt(pump_cal, delimiter="\t")
+	diluted = []
+	efficiency = []
+	last = []
+
+	for vial in vial_count:
+		pump_dir = evolver_dir + "/%s/%s/pump_log/vial%s_pump_log.txt" % (expt_subdir[0], experiment, vial)
+		ODset_dir = evolver_dir + "/%s/%s/ODset/vial%s_ODset.txt" % (expt_subdir[0], experiment, vial)
+		data = np.genfromtxt(pump_dir, delimiter=',', skip_header=2)
+
+		dil_triggered = len(data)
+
+		if dil_triggered != 0:
+			volume = str(round(sum(data[:, 1]) * cal[0, vial] / 1000, 2))
+
+			dil_intervals = len(np.genfromtxt(ODset_dir, delimiter=",", skip_header=2)) / 2
+			extra_dils = dil_triggered - dil_intervals
```

Comment:
What happens in a chemostat condition?

Reply:
I assume

Reply:
I haven't used the chemostat condition, so I am not sure how the data is logged. I assume the pump log is the same. In that case, media consumption calculation would work too, but pump efficiency cannot, since the dilution is constant and not based on a target OD. Am I right?

I will update the code accordingly.

Reply:
That sounds right. Maybe this should only be displayed if a t-stat like experiment is being run, could be a flag in the header of the pump_log file? So the DPU experiment code would also need modifications to add this header.

Reply:
That's a good idea! In fact, in chemostat experiments, the file ODSet is not used, it is just created at the beginning and left like this:
```
Experiment: name vial time
0,0
```
So checking if  `dilution_intervals == 0` should work!